File: front/src/models/Book.ts
import { Author } from "./Author";
import { Category } from "./Category";
import { Dimension } from "./Dimension";
import { PricingGroup } from "./PricingGroup";
import { PublishingCompany } from "./PublishingCompany";

  export interface IBookModel {
    id?: number;
    title?: string;
    author?: Author;
    categories?: Category[];
    year?: string;
    publishingCompany?: PublishingCompany;
    acquisitionValue?: string;
    edition?: string;
    isbn?: string;
    pages?: string;
    synopsis?: string;
    dimensions?: Dimension;
    pricingGroup?: PricingGroup;
    barcode?: string;
    active?: boolean;
  }
  
  export class Book {
    id: number;
    title: string;
    author: Author;
    categories: Category[];
    year: string;
    publishingCompany: PublishingCompany;
    acquisitionValue: string;
    edition: string;
    isbn: string;
    pages: string;
    synopsis: string;
    dimensions: Dimension;
    pricingGroup: PricingGroup;
    barcode: string;
    active: boolean;
  
    constructor({
      id = 0,
      title = '',
      author = (new Author()),
      categories = [],
      year = '',
      publishingCompany = (new PublishingCompany()),
      acquisitionValue = '',
      edition = '',
      isbn = '',
      pages = '',
      synopsis = '',
      dimensions = (new Dimension()),
      pricingGroup = (new PricingGroup()),
      barcode = '',
      active= true
    }: IBookModel = {}) {
      this.id = id;
      this.title = title;
      this.author = author;
      this.categories = categories;
      this.year = year;
      this.publishingCompany = publishingCompany;
      this.acquisitionValue = acquisitionValue;
      this.edition = edition;
      this.isbn = isbn;
      this.pages = pages;
      this.synopsis = synopsis;
      this.dimensions = dimensions;
      this.pricingGroup = pricingGroup;
      this.barcode = barcode;
      this.active = active;
    }

    // get id(): number {
    //   return this._id;
    // }
    // set id(id: number) {
    //   this._id = id;
    // }
  
    // get title(): string {
    //   return this._title;
    // }
    // set title(title: string) {
    //   this._title = title;
    // }
  
    // get author(): Author {
    //   return this._author;
    // }
    // set author(author: Author) {
    //   this._author = author;
    // }
  
    // get categories(): Category[] {
    //   return this._categories;
    // }
    // set categories(categories: Category[]) {
    //   this._categories = categories;
    // }
  
    // get year(): string {
    //   return this._year;
    // }
    // set year(year: string) {
    //   this._year = year;
    // }
  
    // get publishingCompany(): PublishingCompany {
    //   return this._publishingCompany;
    // }
    // set publishingCompany(publishingCompany: PublishingCompany) {
    //   this._publishingCompany = publishingCompany;
    // }
  
    // get acquisitionValue(): string {
    //   return this._acquisitionValue;
    // }
    // set acquisitionValue(acquisitionValue: string) {
    //   this._acquisitionValue = acquisitionValue;
    // }
  
    // get edition(): string {
    //   return this._edition;
    // }
    // set edition(edition: string) {
    //   this._edition = edition;
    // }
  
    // get isbn(): string {
    //   return this._isbn;
    // }
    // set isbn(isbn: string) {
    //   this._isbn = isbn;
    // }
  
    // get pages(): string {
    //   return this._pages;
    // }
    // set pages(pages: string) {
    //   this._pages = pages;
    // }
  
    // get synopsis(): string {
    //   return this._synopsis;
    // }
    // set synopsis(synopsis: string) {
    //   this._synopsis = synopsis;
    // }
  
    // get dimensions(): Dimension {
    //   return this._dimensions;
    // }
    // set dimensions(dimensions: Dimension) {
    //   this._dimensions = dimensions;
    // }
  
    // get pricingGroup(): PricingGroup {
    //   return this._pricingGroup;
    // }
    // set pricingGroup(pricingGroup: PricingGroup) {
    //   this._pricingGroup = pricingGroup;
    // }
  
    // get barcode(): string {
    //   return this._barcode;
    // }
    // set barcode(barcode: string) {
    //   this._barcode = barcode;
    // }

    // get active(): boolean {
    //     return this._active;
    // }
    // set active(active: boolean) {
    //     this._active = active;
    // }

    // toJSON(): IBookModel {
    //   return {
    //     id: this._id,
    //     title: this._title,
    //     author: this._author,
    //     categories: this._categories,
    //     year: this._year,
    //     publishingCompany: this._publishingCompany,
    //     acquisitionValue: this._acquisitionValue,
    //     edition: this._edition,
    //     isbn: this._isbn,
    //     pages: this._pages,
    //     synopsis: this._synopsis,
    //     dimensions: this._dimensions,
    //     pricingGroup: this._pricingGroup,
    //     barcode: this._barcode,
    //     active: this._active,
    //   };
    // }
  }
  

File: back/src/main/java/com/nrabello/back/domain/Book.java
package com.nrabello.back.domain;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "Livros")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="liv_id")
    private Long id;
    @Column(name="liv_titulo")
    private String title;
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name="liv_aut_id", referencedColumnName = "aut_id")
    private Author author;
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "Livros_Categorias",
            joinColumns = @JoinColumn(name = "liv_id"),
            inverseJoinColumns = @JoinColumn(name = "cat_id"))
    private List<Category> categories = new ArrayList<>();
    @Column(name="liv_ano")
    private String year;
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name="liv_edi_id", referencedColumnName = "edi_id")
    private PublishingCompany publishingCompany;
    @Column(name="liv_valorAquisicao")
    private String acquisitionValue;
    @Column(name="liv_edicao")
    private String edition;
    @Column(name="liv_isbn")
    private String isbn;
    @Column(name="liv_paginas")
    private Integer pages;
    @Column(name="liv_sinopse")
    private String synopsis;
    @OneToOne(mappedBy = "book",fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Dimension dimensions;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="liv_grp_id", referencedColumnName = "grp_id")
    private PricingGroup pricingGroup;
    @Column(name="liv_codigoBarra")
    private String barcode;
    @Column(name="liv_ativo")
    private Boolean active;

    public Book() {
    }

    public Book(Long id, String title, Author author, String acquisitionValue, String synopsis, List<Category> categories, String year, PublishingCompany publishingCompany, String edition, String isbn, Integer pages, Dimension dimensions, PricingGroup pricingGroup, String barcode,Boolean active) {
        this.setId(id);
        this.setTitle(title);
        this.setAuthor(author);
        this.setAcquisitionValue(acquisitionValue);
        this.setSynopsis(synopsis);
        this.setCategories(categories);
        this.setYear(year);
        this.setPublishingCompany(publishingCompany);
        this.setEdition(edition);
        this.setIsbn(isbn);
        this.setPages(pages);
        this.setDimensions(dimensions);
        this.setPricingGroup(pricingGroup);
        this.setBarcode(barcode);
        this.setActive(active);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Author getAuthor() {
        return author;
    }

    public void setAuthor(Author author) {
        this.author = author;
    }

    public String getAcquisitionValue() {
        return acquisitionValue;
    }

    public void setAcquisitionValue(String acquisitionValue) {
        this.acquisitionValue = acquisitionValue;
    }

    public String getSynopsis() {
        return synopsis;
    }

    public void setSynopsis(String synopsis) {
        this.synopsis = synopsis;
    }

    public List<Category> getCategories() {
        return categories;
    }

    public void setCategories(List<Category> categories) {
        this.categories = categories;
    }

    public String getYear() {
        return year;
    }

    public void setYear(String year) {
        this.year = year;
    }

    public PublishingCompany getPublishingCompany() {
        return publishingCompany;
    }

    public void setPublishingCompany(PublishingCompany publishingCompany) {
        this.publishingCompany = publishingCompany;
    }

    public String getEdition() {
        return edition;
    }

    public void setEdition(String edition) {
        this.edition = edition;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public Integer getPages() {
        return pages;
    }

    public void setPages(Integer pages) {
        this.pages = pages;
    }

    public Dimension getDimensions() {
        return dimensions;
    }

    public void setDimensions(Dimension dimensions) {
        this.dimensions = dimensions;
    }

    public PricingGroup getPricingGroup() {
        return pricingGroup;
    }

    public void setPricingGroup(PricingGroup pricingGroup) {
        this.pricingGroup = pricingGroup;
    }

    public String getBarcode() {
        return barcode;
    }
    public void setBarcode(String barcode) {
        this.barcode = barcode;
    }
    public Boolean getActive() {
        return active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }
}
